Trying to provide access to STIR data via a Python array

evgueni-ovtchinnikov opened the pull request:
<!-- Fill in most of this text, and delete what is not appropriate.
Please read and adhere to the [contribution guidelines](https://github.com/UCL/STIR/blob/master/CONTRIBUTING.md).
Did you sign the STIR Contribution License Agreement?
-->

## Changes in this pull request


## Testing performed


## Related issues
<!-- Use keywords such as "fixes", "closes", see https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue -->


## Checklist before requesting a review
<!--Put an x between the [] when completed. Delete a line if not applicable. -->
  - [] I have performed a self-review of my code
  - [] I have added docstrings/doxygen in line with the guidance in the developer guide
  - [] I have implemented unit tests that cover any new or modified functionality (if applicable)
  - [x] The code builds and runs on my machine
  - [] `documentation/release_XXX.md` has been updated with any functionality change (if applicable)

KrisThielemans commented:
No, this cannot work. The check is there to make sure that access to the pointer is safe. If the data is not contiguous, accessing data through the pointer will cause segfaults.

The question is why the data is not contiguous in the first place. There is #1468. Could you add your use-case in that issue?

evgueni-ovtchinnikov commented:
@KrisThielemans @casperdcl I applied my _temporary_ quick-fix so that Casper could try to use in Python the `C++` address returned by `ImageData.address`.

@casperdcl this is a simple test I did myself (having applied my quick-fix):

in `stir_data_containers.h` I replaced
```
		size_t address() {
		    return reinterpret_cast<size_t>(_data->get_full_data_ptr());
		}
```
with
```
		size_t address() {
		    float* ptr = _data->get_full_data_ptr();
		    std::cout << (size_t)ptr << '\n';
		    std::cout << *ptr << '\n';
		    std::cout << *(ptr + 1) << '\n';
		    return reinterpret_cast<size_t>(ptr);
		}
```
and in `PET/acquisition_data.py` I added
```
    img_data = acq_data.create_uniform_image(2.0)
    ptr = img_data.address
    print(ptr)
    img_arr = img_data.as_array()
    print(img_arr[0,0,0])
    print(img_arr[0,0,1])
    exit()
```
after line 83. The output from C++ and Python coincided (@casperdcl better fill `img_data` with 1,2,3,... and check the contiguity of the data directly in `address()` - I was too tired to do this on Friday evening, sorry).
